### PR TITLE
[DRAFT] Pregnancy and oviposition mechanics

### PR DESCRIPTION
**Pull request by ChungusGamer666**
# About The Pull Request

Read https://github.com/SPLURT-Station/S.P.L.U.R.T-Station-13/issues/299

## Why It's Good For The Game

image

## A Port?

No.

## Pre-Merge Checklist
- [x] You tested this on a local server.
- [ ] This code did not runtime during testing.
- [ ] You documented all of your changes.

## Changelog

:cl:
add: Pregnancy and oviposition
/:cl:

### CODE REVIEW

**Comment by Mannybrado** [comment]
happy pr # 300! 🥳

File: modular_eros/code/datums/status_effects/pregnancy.dm [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+	if(!QDELETED(mommy))
+		mommy_name = mommy.real_name
+
+	to_chat(babby, "You are the son (or daughter) of [mommy_name]!")
+	var/time = world.time
+	var/name = input(babby, "What will be your name? (You have a minute to decide!)", "Name yourself") as null|text
+	if(world.time - time >= 1 MINUTES)
+		to_chat(babby, "You took too long to decide a name and have been given a random name instead.")
+	if(!name || (world.time - time >= 1 MINUTES))
+		babby.real_name = random_unique_name(babby.gender, FALSE)
```
Comment (ondrej008) [inline modular_eros/code/datums/status_effects/pregnancy.dm:234]: Shouldn't the baby's surname be the same as the mother/father?

File: modular_splurt/code/modules/arousal/arousal.dm [hunk 2]
```diff
@@ -5,6 +5,17 @@
...
 	. = ..()
 
+	//gregnancy...
+	if(!spill && istype(source_gen, /obj/item/organ/genital/penis) && \
+		istype(receiver, /obj/item/organ/genital/vagina) && prob(PREGNANCY_CHANCE_ON_EJACULATION))
+		var/can_impregnate = TRUE
+		if(source_gen.owner.client?.prefs)
+			can_impregnate = source_gen.owner.client.prefs.virility
+		var/can_get_pregnant = (client?.prefs?.fertility && !is_type_in_typecache(src.type, GLOB.pregnancy_blocked_mob_typecache))
+		if(can_impregnate && can_get_pregnant)
```
Comment (ondrej008) [inline modular_splurt/code/modules/arousal/arousal.dm:15]: Doesn't take into account contraceptives (though I guess there aren't any).

Comment (trollertroll) [reply modular_splurt/code/modules/arousal/arousal.dm:15]: Sus

**Comment by trollertroll** [comment]
🤓

**Comment by checkraisefold** [comment]
So why do you need a fully realistic pregnancy system in a spessman game?

**Comment by BongaTheProto** [comment]
> So why do you need a fully realistic pregnancy system in a spessman game?

why not?

**Comment by carlopault** [comment]
> > So why do you need a fully realistic pregnancy system in a spessman game?
> 
> why not?

Feature creep and because it was only made to satisfy another disgusting fetish.

File: modular_splurt/code/modules/arousal/arousal.dm (hunk 2 above)
Comment (ChungusGamer666) [reply modular_splurt/code/modules/arousal/arousal.dm:15]: Will take contraceptives into account when i add them.

File: modular_eros/code/datums/status_effects/pregnancy.dm (hunk 1 above)
Comment (ChungusGamer666) [reply modular_eros/code/datums/status_effects/pregnancy.dm:234]: Will change this so the mother decides the name, if possible, actually.

**Comment by ChungusGamer666** [comment]
> Feature creep and because it was only made to satisfy another disgusting fetish.
Absolutely but who cares this server is made for disgusting fetish content and i am being paid

**Comment by checkraisefold** [comment]
> > So why do you need a fully realistic pregnancy system in a spessman game?
> 
> why not?

The Industrial Revolution and its consequences have been a disaster for the human race. They have greatly increased the life-expectancy of those of us who live in “advanced” countries, but they have destabilized society, have made life unfulfilling, have subjected human beings to indignities, have led to widespread psychological suffering (in the Third World to physical suffering as well) and have inflicted severe damage on the natural world. The continued development of technology will worsen the situation. It will certainly subject human beings to greater indignities and inflict greater damage on the natural world, it will probably lead to greater social disruption and psychological suffering, and it may lead to increased physical suffering even in “advanced” countries.

**Comment by windows-fryer** [comment]
You see batman... A person uploads a pregnancy pull request and no one bats an eye, but when I say care for yourself batman, society goes wild....

**Comment by Fajetti** [comment]
Alright guys, place your bets on how long until there's a new rule on the server that says "You cannot ERP with babies, even though they're not player controlled"

I'm saying either a few rounds to never.

**Comment by carlopault** [comment]
> > Feature creep and because it was only made to satisfy another disgusting fetish.
> > Absolutely but who cares this server is made for disgusting fetish content and i am being paid

How much could they have possibly paid you to make this

**Comment by BongaTheProto** [comment]
> > > Feature creep and because it was only made to satisfy another disgusting fetish.
> > > Absolutely but who cares this server is made for disgusting fetish content and i am being paid
> 
> How much could they have possibly paid you to make this

It clearly states it's 100 dollars

**Comment by restlessObserver** [comment]
So how it's going..

**Comment by ChungusGamer666** [comment]
https://github.com/SPLURT-Station/S.P.L.U.R.T-Station-13/issues/299 has been severely complicated and i have to start over COMPLETELY

**Comment by BongaTheProto** [comment]
> I have a couple of concerns. Are you planning to have an object of a newborn and how would it be safe from any **Dubious** players doing unholy shit imaginable? I can see Oviposition be safe as it's just eggs but I _really_ don't like the idea of there being any offspring spawning in the game.
> 
> I can see a solution being just having them be bluespace to a nursery off station without any issue. Might be good rp for doctors but honestly tho, I don't see many players enjoying the idea of helping someone give birth.
> 
> The impregnation feature is a cool idea but I just want to advise there be a lot of consideration for it to be comfortable as possible for other players.

there are no actual babies involved. Read the bounty text

**Comment by C0rva1r** [comment]
> > I have a couple of concerns. Are you planning to have an object of a newborn and how would it be safe from any **Dubious** players doing unholy shit imaginable? I can see Oviposition be safe as it's just eggs but I _really_ don't like the idea of there being any offspring spawning in the game.
> > I can see a solution being just having them be bluespace to a nursery off station without any issue. Might be good rp for doctors but honestly tho, I don't see many players enjoying the idea of helping someone give birth.
> > The impregnation feature is a cool idea but I just want to advise there be a lot of consideration for it to be comfortable as possible for other players.
> 
> there are no actual babies involved. Read the bounty text

Oooh ok, my bad. Disregard what I mentioned before then.

**Pull request closed without merging**